core/spring-boot/src/main/java/org/springframework/boot/logging/log4j2/Log4J2RuntimeHints.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2025-present the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.logging.log4j2;
+
+import org.springframework.aot.hint.RuntimeHints;
+import org.springframework.aot.hint.RuntimeHintsRegistrar;
+import org.springframework.util.ClassUtils;
+
+/**
+ * {@link RuntimeHintsRegistrar} implementation for {@link Log4J2LoggingSystem}.
+ *
+ * @author Piotr P. Karwasz
+ */
+class Log4J2RuntimeHints implements RuntimeHintsRegistrar {
+
+	// Log4j2LoggingSystem checks for the presence of these classes reflectively.
```

Comment:
We don't write comments like this. If the property moves to the logging system, we can then add some field javadoc to describe why it is used.

Reply:
Fixed in https://github.com/spring-projects/spring-boot/pull/46410/commits/9139fe96d5797611bf37933f4d80bdde082c9164